src/core/cardano/walletConnect/identityWalletConnect.ts

```diff
@@ -45,15 +53,41 @@ class IdentityWalletConnect extends CardanoPeerConnect {
 
     this.sign = async (
       identifier: string,
-      payload: string
+      payload: string,
+      onSignError?: (error: Error) => void
     ): Promise<string> => {
-      if (this.signerCache.get(identifier) === undefined) {
-        this.signerCache.set(
+      let approved: boolean | undefined = undefined;
+      // Closure that updates approved variable
+      const approvalCallback = (approvalStatus: boolean) => {
+        approved = approvalStatus;
+      };
+      this.eventService.emit<PeerConnectSigningEvent>({
+        type: PeerConnectSigningEventTypes.PeerConnectSign,
+        payload: {
           identifier,
-          await Agent.agent.identifiers.getSigner(identifier)
-        );
+          payload,
+          approvalCallback,
+        },
+      });
+      // Wait until approved is true or false
+      while (approved === undefined) {
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+      }
+      if (approved) {
+        if (this.signerCache.get(identifier) === undefined) {
+          this.signerCache.set(
+            identifier,
+            await Agent.agent.identifiers.getSigner(identifier)
+          );
+        }
+        return this.signerCache.get(identifier)!.sign(Buffer.from(payload))
+          .qb64;
+      } else {
+        if (typeof onSignError === "function") {
+          onSignError(new Error("User declined the signing request"));
```

Comment:
Lets wait until we know more from Fabian but `onSignError` seems to be a function in cardano-ballot, and not cardano peer connect. Not sure though.